File: skeleton/src/main/java/me/shkschneider/skeleton/ui/transforms/RotateDownTransformer.java
package me.shkschneider.skeleton.ui.transforms;

import android.view.View;

// <https://github.com/ToxicBakery/ViewPagerTransforms>
public class RotateDownTransformer extends ABaseTransformer {

	private static final float ROT_MOD = -15F;

	@Override
	protected void onTransform(final View view, final float position) {
		final float width = view.getWidth();
		final float height = view.getHeight();
		final float rotation = ROT_MOD * position * -1.25F;
		view.setPivotX(width * 0.5F);
		view.setPivotY(height);
		view.setRotation(rotation);
	}
	
	@Override
	protected boolean isPagingEnabled() {
		return true;
	}

}

File: library/src/main/java/shkschneider/skeleton/ui/transforms/RotateDownTransformer.kt
package me.shkschneider.skeleton.ui.transforms

import android.view.View

// <https://github.com/ToxicBakery/ViewPagerTransforms>
class RotateDownTransformer : ABaseTransformer() {

    private val ROT_MOD = -1.toFloat()

    override fun isPagingEnabled(): Boolean {
        return true
    }

    override fun onTransform(page: View, position: Float) {
        val width = page.width.toFloat()
        val height = page.height.toFloat()
        val rotation = ROT_MOD * position * -1.1.toFloat()
        page.pivotX = width * 0.1.toFloat()
        page.pivotY = height
        page.rotation = rotation
    }

}
